### PR TITLE
Add heading and improve docs

### LINKED ISSUE #17: Add Ecto heading
Add module `ExProsemirror.Heading` that create ecto embedded_schema.

Heading should be a block by itself

### PR DESCRIPTION
**Pull request by Matsa59**
## 📖 Description

Add ecto schema for heading and update some docs.
Another PR will follows to improve code architcture and docs.

Close #17 

## 📋 Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### CODE REVIEW

File: lib/ex_prosemirror/doc.ex [hunk 1]
```diff
@@ -1,18 +1,30 @@
 defmodule ExProsemirror.Doc do
-  @moduledoc false
+  @moduledoc ~S"""
+  Root of prosemirror element. It handle element form input and cast it to Ecto schemas.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/doc.ex:3]:
```suggestion
  Root of prosemirror element. It handles element form input and casts it to Ecto schemas.
```

File: lib/ex_prosemirror/doc.ex [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
+
+  A doc could contains one or many :
```
Comment (tomkonidas) [inline lib/ex_prosemirror/doc.ex:5]:
```suggestion
  A doc could contain one or many :
```

**Review by onixus74 — APPROVED** [review]

File: lib/ex_prosemirror/html/form.ex [hunk 3]
```diff
@@ -10,8 +10,9 @@ defmodule ExProsemirror.HTML.Form do
 
   ## Options
 
-  - marks: mark tags to use defined in your ProseMirror config
-  - blocks: block tags to use defined in your ProseMirror config
+  - `marks`: mark tags to use defined in your ProseMirror config
+  - `blocks`: block tags to use defined in your ProseMirror config
+  - `type`: Type of input to use (see: `ExProsemirror.Config` for more informations)
```
Comment (tomkonidas) [inline lib/ex_prosemirror/html/form.ex:15]:
```suggestion
  - `type`: type of input to use (see: `ExProsemirror.Config` for more informations)
```

All the other ones do not begin with a capital, so I would make them all capital or none

File: lib/ex_prosemirror/html/form.ex [hunk 4]
```diff
@@ -24,16 +25,20 @@ defmodule ExProsemirror.HTML.Form do
 
   ## Usages
 
-      <%= prosemirror_input @form, :title, id: "my-article-input", blocks: [heading: [1]] %>
-      <%= prosemirror_input @form, :body, id: "my-article-input", blocks: [:p, heading: [2, 3]] %>
-      <%= prosemirror_input @form, :author, id: "my-article-input", marks: [:strong], blocks: [:p] %>
+      <%= prosemirror_input @form, :title, id: "my-article-input" %>
+      # Generate prosemirror input with :default configuration.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/html/form.ex:29]:
```suggestion
      # Generates a prosemirror input with the :default configuration.
```

File: lib/ex_prosemirror/html/form.ex [hunk 5]
```diff
@@ -24,16 +25,20 @@ defmodule ExProsemirror.HTML.Form do
...
+
+      <%= prosemirror_input @form, :title, type: :title, id: "my-article-input" %>
+      # Generate prosemirror input with :title configuration.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/html/form.ex:32]:
```suggestion
      # Generates a prosemirror input with the :title configuration.
```

File: lib/ex_prosemirror/html/form.ex [hunk 6]
```diff
@@ -24,16 +25,20 @@ defmodule ExProsemirror.HTML.Form do
...
+
+      <%= prosemirror_input @form, :title, type: :title, id: "my-article-input", block: [h1: false] %>
+      # Override the :title configuration to disable h1 node.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/html/form.ex:35]:
```suggestion
      # Overrides the :title configuration to disable the h1 node.
```

File: lib/ex_prosemirror/html/form.ex [hunk 7]
```diff
@@ -24,16 +25,20 @@ defmodule ExProsemirror.HTML.Form do
...
+
+      <%= prosemirror_input @form, :title, type: :title, id: "my-article-input", block: [h2: true] %>
+      # Override the :title configuration to enable h2 node.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/html/form.ex:38]:
```suggestion
      # Overrides the :title configuration to enable the h2 node.
```

File: lib/ex_prosemirror/html/form.ex [hunk 8]
```diff
@@ -24,16 +25,20 @@ defmodule ExProsemirror.HTML.Form do
...
 
-  ## Options
 
-  - `blocks`:
-    - `:heading`: allow headers. Should be an array of integer (value between 1 to 6).
-    - `:p`: allow paragraph.
-  - `marks`: Could be `strong` or `em`.
+  Read `ExProsemirror.Config` for more informations about configuration.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/html/form.ex:41]:
```suggestion
  Read `ExProsemirror.Config` for more information about configuration.
```

File: lib/ex_prosemirror/text.ex [hunk 9]
```diff
@@ -1,20 +1,40 @@
 defmodule ExProsemirror.Text do
-  @moduledoc false
+  @moduledoc ~S"""
+  Prosemirror inliner element. It's contained by most of ExProsemirror modifiers like
```
Comment (tomkonidas) [inline lib/ex_prosemirror/text.ex:3]:
```suggestion
  Prosemirror inline element. It's contained by most of ExProsemirror modifiers like
```

File: lib/ex_prosemirror/text.ex [hunk 10]
```diff
@@ -1,20 +1,40 @@
...
+  `ExProsemirror.Paragraph` or `ExProsemirror.Heading`.
+
+  When creating a ExProsemirror node you can use this module thanks to
```
Comment (tomkonidas) [inline lib/ex_prosemirror/text.ex:6]:
```suggestion
  When creating an ExProsemirror node you can use this module thanks to
```

File: lib/ex_prosemirror/text.ex [hunk 11]
```diff
@@ -1,20 +1,40 @@
...
+
+      embedded_prosemirror_field([text: ExProsemirror.Text], array: true)
+
+  > `embedded_prosemirror_field` is define in `ExProsemirror.SchemaHelper` and auto import whe
```
Comment (tomkonidas) [inline lib/ex_prosemirror/text.ex:10]:
```suggestion
  > `embedded_prosemirror_field` is defined in `ExProsemirror.SchemaHelper` and auto imports with
```

File: lib/ex_prosemirror/text.ex [hunk 12]
```diff
@@ -1,20 +1,40 @@
...
   end
 
+  @doc false
   def changeset(struct_or_changeset, attrs \\ %{}) do
     struct_or_changeset
     |> cast(attrs, [:text])
   end
 
+  @doc ~S"""
+  Return the text of a `ExProsemirror.Text`.
```
Comment (tomkonidas) [inline lib/ex_prosemirror/text.ex:32]:
```suggestion
  Returns the text of an `ExProsemirror.Text`.
```

**Review by tomkonidas — APPROVED** [review]

**Pull request merged**